yarn-project/simulator/src/private/acvm/oracle/oracle.ts

```diff
@@ -452,4 +459,31 @@ export class Oracle {
       frToNumber(fromACVMField(numEntries)),
     );
   }
+
+  async aes128Decrypt(
+    ciphertextBVecStorage: ACVMField[],
+    [ciphertextLength]: ACVMField[],
+    iv: ACVMField[],
+    symKey: ACVMField[],
+  ): Promise<(ACVMField | ACVMField[])[]> {
+    const ciphertext = fromUintBoundedVec(ciphertextBVecStorage, ciphertextLength, 8);
+    const ivBuffer = fromUintArray(iv, 8);
+    const symKeyBuffer = fromUintArray(symKey, 8);
+
+    const plaintext = await this.typedOracle.aes128Decrypt(ciphertext, ivBuffer, symKeyBuffer);
+    return bufferToBoundedVec(plaintext, ciphertextBVecStorage.length);
+  }
+
+  async getSharedSecret(
+    [address]: ACVMField[],
+    [ephPKField0]: ACVMField[],
+    [ephPKField1]: ACVMField[],
+    [ephPKField2]: ACVMField[],
```

Comment:
When I had the ephPk input defined only as "ephPk: ACVMField[]" then ephPk was only 1 Field instead of 3. Does the reviewer if it's really necessary to list the components directly here? It's ugly.

Reply:
I'm not sure I follow. It's a struct with 3 values so it gets converted to three acvm fields no?

Reply:
If I put there ephPk: ACVField[] instead of the 3 components I got just one field.